examples/basic/ui/dialog.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+
+from bokeh.io import show
+from bokeh.models import Button, Column, CustomJS, Dialog, OpenDialog
+from bokeh.plotting import figure
+
+
+def plot(N: int):
+    x = np.random.random(size=N) * 100
+    y = np.random.random(size=N) * 100
+    radii = np.random.random(size=N) * 1.5
+    colors = np.array([(r, g, 150) for r, g in zip(50+2*x, 30+2*y)], dtype="uint8")
+
+    p = figure(active_scroll="wheel_zoom", lod_threshold=None, title=f"Plot with N={N} circles", sizing_mode="stretch_both")
+    p.circle(x, y, radius=radii, fill_color=colors, fill_alpha=0.6, line_color=None)
+
+    return p
+
+show_plot = Button(label="Show plot ...")
+
+close_dialog = Button(label="Close")
+dialog = Dialog(
+    title="Dialog with a plot",
+    content=Column(
+        sizing_mode="stretch_both",
+        children=[
+            plot(500),
+            close_dialog,
+        ],
+    ),
+)
+show_plot.js_on_click(OpenDialog(dialog=dialog))
+close_dialog.js_on_click(
```

Comment:
Why is it not symmetric with `OpenDialog`?

Reply:
If we we are good with this kind of API, then I will fill in the missing parts. Though it's also good to show how to do things manually, if an API is missing.

Reply:
There seems to be no `CloseDialog` equivalent, correct?

Running this example with Bokeh serve (and updating the code with `show` -> `curdoc().add_root`, I can't get the close button to work and get the log:
``` console
Uncaught (in promise) Error: cannot find a view for Dialog(p1046)
    get_one http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:221
    default blob:http://localhost:5006/4e8e6ce5-4530-4159-ae07-e228cfae3dcf:1
    execute http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:480
    execute http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:216
    _process_event http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:217
    trigger http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:166
    trigger_event http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:217
    click http://localhost:5006/static/js/bokeh-widgets.min.js?v=8945a2a59d802de388cad0ff943167d1f705d69f095c16e44e908e94b6dd70ceba9b48c6ab0601f0d6b1c593988520f8ab2ce507540022d08cc8888dccb244d8:57
    render http://localhost:5006/static/js/bokeh-widgets.min.js?v=8945a2a59d802de388cad0ff943167d1f705d69f095c16e44e908e94b6dd70ceba9b48c6ab0601f0d6b1c593988520f8ab2ce507540022d08cc8888dccb244d8:47
    render http://localhost:5006/static/js/bokeh-widgets.min.js?v=8945a2a59d802de388cad0ff943167d1f705d69f095c16e44e908e94b6dd70ceba9b48c6ab0601f0d6b1c593988520f8ab2ce507540022d08cc8888dccb244d8:47
    render http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:573
    render_to http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:573
    render http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:486
    _toggle http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:486
    connect_signals http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:486
    n http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:221
    emit http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:181
    emit http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:181
    _setv http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:180
    setv http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:180
    open http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:486
    execute http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:485
    execute http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:216
    _process_event http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:217
    trigger http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:166
    trigger_event http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:217
    click http://localhost:5006/static/js/bokeh-widgets.min.js?v=8945a2a59d802de388cad0ff943167d1f705d69f095c16e44e908e94b6dd70ceba9b48c6ab0601f0d6b1c593988520f8ab2ce507540022d08cc8888dccb244d8:57
    render http://localhost:5006/static/js/bokeh-widgets.min.js?v=8945a2a59d802de388cad0ff943167d1f705d69f095c16e44e908e94b6dd70ceba9b48c6ab0601f0d6b1c593988520f8ab2ce507540022d08cc8888dccb244d8:47
    render http://localhost:5006/static/js/bokeh-widgets.min.js?v=8945a2a59d802de388cad0ff943167d1f705d69f095c16e44e908e94b6dd70ceba9b48c6ab0601f0d6b1c593988520f8ab2ce507540022d08cc8888dccb244d8:47
    render_to http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:573
    s http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:220
    s http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:220
    add_document_standalone http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:220
    add_document_from_session http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:232
    k http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:164
    embed_items http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:164
    embed_document http://localhost:5006/dialog:38
    fn http://localhost:5006/dialog:41
    fn http://localhost:5006/dialog:57
    safely http://localhost:5006/static/js/bokeh.min.js?v=09b616ff6e78e0bc7aa27d971d6bd254ee707011ace76821753ce252849fa5443f279823051efd10b5ca74b0a9a148a1306310dbde3a3dcae097acca168ae1bf:240
    fn http://localhost:5006/dialog:33
    EventListener.handleEvent* http://localhost:5006/dialog:61
    <anonymous> http://localhost:5006/dialog:62
bokeh.min.js:221:4067
```

Reply:
> Error: cannot find a view for Dialog(p1046)

This actually doesn't work at all. This is because `OpenDialog` (callbacks in general) doesn't have a view, so it doesn't participate in the mechanism that allows to resolve views via the global view manager. Given the intent behind `OpenDialog` and this example is to have a non-root, detached/global dialog, then I need to add a mechanism to register such "free" views.